core/src/test/java/org/elasticsearch/action/index/IndexResponseTests.java

```diff
@@ -0,0 +1,153 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.index;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.action.support.replication.ReplicationResponse;
+import org.elasticsearch.action.support.replication.ReplicationResponseTests;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.util.CollectionUtils;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class IndexResponseTests extends ESTestCase {
+
+    public void testIndexResponseToXContent() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+        final IndexResponse indexResponse = randomIndexResponse();
+        final BytesReference indexResponseBytes = XContentHelper.toXContent(indexResponse, xContentType, true);
+
+        Map<String, Object> sourceAsMap = XContentHelper.convertToMap(indexResponseBytes, false).v2();
+        assertIndexResponse(indexResponse, sourceAsMap, false);
+    }
+
+    public void testIndexResponseFromXContent() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+
+        // Create a random IndexResponse and converts it to XContent in bytes
+        final IndexResponse indexResponse = randomIndexResponse();
+        final BytesReference indexResponseBytes = XContentHelper.toXContent(indexResponse, xContentType, true);
+
+        // Parse the XContent bytes to obtain a parsed IndexResponse
+        IndexResponse parsedIndexResponse;
+        try (XContentParser parser = xContentType.xContent().createParser(indexResponseBytes)) {
+            parsedIndexResponse = IndexResponse.fromXContent(parser);
+            assertNull(parser.nextToken());
+        }
+
+        // Finally convert the parsed IndexResponse to XContent...
+        BytesReference parsedIndexResponseBytes = XContentHelper.toXContent(parsedIndexResponse, xContentType, true);
+        // ... and convert the result to a Map
+        Map<String, Object> parsedSourceAsMap = XContentHelper.convertToMap(parsedIndexResponseBytes, false).v2();
```

Comment:
rather than using a map here, couldn't you call fromXContent again? would that make sense?